### DISCUSSION TITLE
Duplicate dimension names for easy data subsetting

### DISCUSSION THREAD

**Discussion by MeyerBender** [pydata/xarray#9603] [Q&A]
Hi,

I want to store an adjacency matrix in an xarray object, which should have the same dimensions at the x and y coordinate. Here is a minimal example:

```
import numpy as np
import xarray as xr

cells = ['cell1', 'cell2', 'cell3']

adjacency_matrix = np.array([
    [0, 1, 0],
    [1, 0, 1],
    [0, 1, 0]
])

da = xr.DataArray(
    adjacency_matrix,
    coords=[cells, cells],
    dims=['cells', 'cells'],
    name='adjacency_matrix'
)
```

Note that with this example, I get the following warning:
```
UserWarning: Duplicate dimension names present: dimensions {'cells'} appear more than once in dims=('cells', 'cells'). We do not yet support duplicate dimension names, but we do allow initial construction of the object. We recommend you rename the dims immediately to become distinct, as most xarray functionality is likely to fail silently if you do not. To rename the dimensions you will need to set the ``.dims`` attribute of each variable, ``e.g. var.dims=('x0', 'x1')``.
```

I know that I could just do `dims=['cells_1', 'cells_2']`. However, I later want to subset my cells, and have this affect both the x and y axis. So something like `da.sel(cells=['cell1', 'cell3'])` should work. The reason why I do not want to distribute this over multiple variables is that my `cells` coordinate actually also appears in some other `DataArrays`, and I want everything to be synchronized.

The subsetting I described above does seem to work correctly, however the warning has me a bit concerned. What are the best practices in this case? Is it fine to ignore the warning, as subsetting on the cells will be my main use case? What are some instances where this duplicate-name approach might fail?

Thank you very much for your help!

**Comment by benbovy** [comment]
One solution for this problem could be to have `split_dims` and `merge_dims` methods as detailed in https://github.com/pydata/xarray/issues/1378#issuecomment-1698651257 so we can easily switch between the two useful representations. It shouldn't be hard to implement I think.
